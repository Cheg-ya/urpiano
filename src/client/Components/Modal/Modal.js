import React, { Component } from 'react';
import './Modal.scss';

class Modal extends Component {
  constructor(props) {
    super(props);
    this.onClickToClose = this.onClickToClose.bind(this);
  }

  onClickToClose(e) {
    if(e.target === e.currentTarget) {
      this.props.closeModal();
    }
  }

  render() {
    return (
      <div className="modalCover" onClick={this.onClickToClose}>
        {this.props.children}
      </div>
    );
  }
}

export default Modal;
