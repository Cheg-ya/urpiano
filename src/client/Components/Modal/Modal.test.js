import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Modal from './Modal';

describe('Modal component test', () => {
  const closeModal = jest.fn().mockImplementation(() => true);
  const component = shallow(<Modal closeModal={closeModal}/>);
  const instance = component.instance();

  it('renders correctly', () => {
    const tree = renderer.create((
      <Modal closeModal={closeModal} />
    )).toJSON();

    expect(tree).toMatchSnapshot();
    expect(component.exists()).toBe(true);
    expect(component.state()).toBeNull();
  });

  it('checks props type', () => {
    expect(typeof instance.props.closeModal).toBe('function');
  });

  it('renders children node when passed in', () => {
    const tree = renderer.create((
      <Modal closeModal={closeModal}>
        <div>Modal Component</div>
      </Modal>
    )).toJSON();

    const component = shallow((
      <Modal closeModal={closeModal}>
        <div>Modal Component</div>
      </Modal>
    ));

    expect(tree).toMatchSnapshot();
    expect(component.contains(<div>Modal Component</div>)).toBe(true);
  });

  it('calls onClickClose method when click event occurred', () => {
    const spy = jest.spyOn(Modal.prototype, 'onClickClose');
    const component = mount(<Modal closeModal={closeModal}></Modal>);

    component.find('.modalCover').simulate('click', { target: true, currentTarget: true });
    expect(spy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledTimes(1);

    component.unmount();
  });

  it('calls props function after onClickClose invoked', () => {
    const component = mount(<Modal closeModal={closeModal}></Modal>);
    const props = component.props();
    const spy = jest.spyOn(props, 'closeModal');

    component.instance().onClickClose({ target: true, currentTarget: true });
    expect(spy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledTimes(1);

    component.unmount();
  });
});
