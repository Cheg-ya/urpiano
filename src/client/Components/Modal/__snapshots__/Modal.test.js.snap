// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component test renders children node when passed in 1`] = `
<div
  className="modalCover"
  onClick={[Function]}
>
  <div>
    Modal Component
  </div>
</div>
`;

exports[`Modal component test renders correctly 1`] = `
<div
  className="modalCover"
  onClick={[Function]}
/>
`;
